### DISCUSSION TITLE
Sunbird-RC usage in Sunbird-ED

### DISCUSSION THREAD

**Discussion by Renuka-S** [Sunbird-Ed/Community#217] [Sunbird ED configuration discussion] [answered]
Hello,

I am looking for the sunbird-rc implementation or usage details in sunbird-ed
- Is sunbird-rc packaged in the sunbird-ed core installation?
- It is mentioned credentials are provided using SB-RC, where can we find implementation details for this? 

@vrayulu @parthlawate @tejash-jl @Pratikshakhandagale

**Comment by parthlawate** [comment]
cc @pramodkvarma

**Comment by vrayulu** [comment, marked as answer]
@rhwarrier @reshmi-nair - please share the details of SB-RC usage in lern

**Reply by reshmi-nair** [reply]
@Renuka-S @parthlawate Please find the details:
Wiki link : https://project-sunbird.atlassian.net/wiki/spaces/SLBB/pages/3092611077/Sunbird+RC+Integration+for+Certificate+Generation

Presentation link: https://docs.google.com/presentation/d/1XcMUl7gJlgtckhn4g8RKdr4GBNEWKj-XLoaI8Ej9qHQ/edit#slide=id.g10f24c18dd2_0_226

**Reply by Renuka-S** [reply]
@reshmi-nair Thank you for the links. Could you please also confirm if the sunbird-RC comes as a core part or need to be installed/maintained separately? cc: @parthlawate

**Reply by vrayulu** [reply]
@gandham-santhosh @keshavprasadms  - please confirm if Sunbird-RC is part of SunbirdEd installation or need to be installed separately. I believe it is the former.

**Reply by beepdot** [reply]
Yes @vrayulu it comes as part of the SunbirdEd installation. But unfortunately those details are missing from the current installation document.
@Renuka-S This is the installation document I am referring to - [SunbirdED Setup](https://ed.sunbird.org/use/prerequisites-for-your-own-sunbird-ed-instance/platform). Can you please let me know if you are referring to the same link?

**Reply by Renuka-S** [reply]
@keshavprasadms Yes. Referring to this one for the installation. Both installation & implementation details are not available in public documentation.

**Reply by beepdot** [reply]
@Renuka-S I am providing you the details on how to set it up as part of Sunbird ED here. The same will be added into the docs later.
These are the jenkins jobs you require - 
https://github.com/project-sunbird/sunbird-devops/tree/master/deploy/jenkins/jobs/Build/jobs/Sunbird-RC/jobs
https://github.com/project-sunbird/sunbird-devops/tree/master/deploy/jenkins/jobs/ArtifactUpload/jobs/dev/jobs/Sunbird-RC/jobs
https://github.com/project-sunbird/sunbird-devops/tree/master/deploy/jenkins/jobs/Deploy/jobs/dev/jobs/Sunbird-RC/jobs

Please take these jobs to your Jenkins VM if its not already available. Once you have the job, you can just run these jobs from [release-4.8.0](https://github.com/project-sunbird/sunbird-devops/releases/tag/release-4.8.0) tag.

#### Build Jobs:
```
Build/Sunbird-RC/CertificateApi
Build/Sunbird-RC/CertificateSign
Build/Sunbird-RC/Registry
```

#### ArtifactUpload Jobs (Auto triggered, please just check that they are successful):
```
ArtifactUpload/{{env}}/Sunbird-RC/CertificateApi
ArtifactUpload/{{env}}/Sunbird-RC/CertificateSign
ArtifactUpload/{{env}}/Sunbird-RC/Registry
```

#### Deploy Jobs:
```
Deploy/{{env}}/Sunbird-RC/UploadRCSchema
Deploy/{{env}}/Sunbird-RC/CertificateApi
Deploy/{{env}}/Sunbird-RC/CertificateSign
Deploy/{{env}}/Sunbird-RC/Registry
```

**Reply by Renuka-S** [reply]
@keshavprasadms Thank you for the details above. Please do share once this is included in the documentation, so we can link in SB-RC docs as well. We will maintain the copy of the above details for now.
Thank you once again.
